float8_experimental/float8_dynamic_linear.py

```diff
@@ -144,13 +144,19 @@ def __new__(cls, tensor: torch.Tensor, mm_config: ScaledMMConfig):
             dtype=tensor.dtype,
             layout=tensor.layout,
             device=tensor.device,
-            pin_memory=tensor.is_pinned(),
+            # TODO: workaround fake tensor not implementing is.pinned
+            # pin_memory=tensor.is_pinned(),
+            pin_memory=False,
             requires_grad=tensor.requires_grad,
         )
 
     def __init__(self, tensor: torch.Tensor, mm_config: ScaledMMConfig):
         self._tensor = tensor
         self._mm_config = mm_config
+        # Optional cache for pre-computed fp8 data/scale
+        self._fp8_data: Optional[torch.Tensor] = None
```

Comment:
One major requirement for tensor subclasses that I don't think is respected here: `__tensor_flatten__` and `__tensor_unflatten__` must properly convey every inner tensor on the subclass.

So when we call `__tensor_flatten__` on this subclass, if either of `_fp8_data/scale/amax` are set to valid tensors, they need to be returned there (and similarly `__tensor_unflatten__` needs to handle them as extra args)

Reply:
thanks for pointing this out! This saves me a lot of debugging time. I can give it a try by including `_fp8_data/scale/amax` in `__tensor_flatten__` and `__tensor_unflatten__`

Reply:
`torch.compile` works after patching https://github.com/pytorch/pytorch/pull/127431 
will compare traces in 2nd PR